### ISSUE TITLE
Singular input to tri_inv / solve_triangular aborts the process

### ISSUE THREAD

**Issue by jessegrabowski** [ml-explore/mlx#4407]
`mx.linalg.tri_inv` and `mx.linalg.solve_triangular` abort when the input is singular. The `std::runtime_error` from `Inverse::eval_cpu` is thrown on a scheduler thread rather than the caller's, so it can't be caught — wrapping the call in `try`/`except` changes nothing and the process exits 134.

```python
import mlx.core as mx

a = mx.array([[1.0, 0.0], [1.0, 0.0]])
mx.eval(mx.linalg.tri_inv(a, upper=False, stream=mx.cpu))
```

```
libc++abi: terminating due to uncaught exception of type std::runtime_error: [Inverse::eval_cpu] triangular inversion failed with error code 2
```

JAX returns NaN for the same input and keeps going, including under `jit`:

```python
import jax
import jax.numpy as jnp
from jax.scipy.linalg import solve_triangular

a = jnp.array([[1.0, 0.0], [1.0, 0.0]])
solve = jax.jit(lambda m, v: solve_triangular(m, v, lower=True))
solve(a, jnp.ones((2, 1)))
# Array([[ 1.], [nan]], dtype=float32)
```

The JAX behavior is preferable in most contexts. Singularity is a runtime property of the input, so an iterative solver can hit it on a trial step it is about to reject anyway — a NaN propagates and the caller recovers, an abort doesn't leave that option. In my case, it killed the interpreter over a step the line search would have discarded.

Environment: mlx 0.32.0, jax 0.11.1, Python 3.12.13, macOS 26.5, arm64.

**Comment by zcbenz** [comment]
The error throws exception instead of aborting after 0.32.1.
